File: sdks/fog/challenges/get_object_metadata.rb
#!/usr/bin/env ruby
require 'fog'

service = Fog::Storage.new({
    :provider             => 'rackspace',
    :rackspace_username   => ENV['RAX_USERNAME'],
    :rackspace_api_key    => ENV['RAX_API_KEY'],
    :rackspace_region     => ENV['RAX_REGION'],
    :rackspace_auth_url   => "#{ENV['RAX_AUTH_URL']}/v2.0"
})
# directory = service.directories.get ENV['TEST_DIRECTORY']
# directory.files.head_object ENV['TEST_FILE']
service.head_object ENV['TEST_DIRECTORY'], ENV['TEST_FILE']

File: sdks/pyrax/challenges/get_object_metadata.py
#!/usr/bin/env python

import os
import pyrax

pyrax.set_http_debug(True)
pyrax.set_setting("region", os.getenv("RAX_REGION"))
pyrax.set_setting("identity_type", "rackspace")
# Create the identity object
# pyrax._create_identity()
# Change its endpoint
pyrax.identity.auth_endpoint = os.getenv('RAX_AUTH_URL') + '/v2.0/'

# Identity Connection - Authenticate
pyrax.set_credentials(os.getenv('RAX_USERNAME'), os.getenv('RAX_API_KEY'))
# Cloud Files API - List Files
cf = pyrax.cloudfiles
cf.get_object_metadata(os.getenv('TEST_DIRECTORY'), os.getenv('TEST_FILE'))
